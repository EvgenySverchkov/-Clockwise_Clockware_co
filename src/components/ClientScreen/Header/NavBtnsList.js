import React from 'react';
import { connect } from "react-redux";
import PropTypes from "prop-types";

import LogOutBtn from "./LogOutBtn";
import LogInBtn from "./LogInBtn";
import NavItem from "./NavItem";

const NavList = ({isAuth}) => (
    <div className="navbar-collapse collapse" id="navbarsExample03">
        <ul className="navbar-nav mr-auto">
            {isAuth ? <LogOutBtn /> : <LogInBtn />}
            <NavItem title = {"SignUp"} link={"/client/registration"}/>
            <NavItem title = {"To main"} link={"/client"}/>
        </ul>
    </div>
);

NavList.propTypes = {
    isAuth: PropTypes.bool
}

function mapStateToProps(state) {
    return {
      isAuth: state.client_order_reduser.isAuth,
    };
}

export default connect(mapStateToProps)(NavList);