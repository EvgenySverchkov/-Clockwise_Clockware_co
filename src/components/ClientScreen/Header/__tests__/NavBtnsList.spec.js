import React from "react";
import {create} from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configMockStore from "redux-mock-store";

import NavBtnsList from "../NavBtnsList";

const mockStore = configMockStore();

describe("Test of <NavBtnsList />", ()=>{
    it("If user is unlogged", ()=>{
        const store = mockStore({
            authReducer:{
                isAuthClient: false
            }
        });
        const component = create(
            <Provider store={store}>
                <MemoryRouter>
                    <NavBtnsList />
                </MemoryRouter>
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("If user is logged", ()=>{
        const store = mockStore({
            authReducer:{
                isAuthClient: true
            }
        });
        const component = create(
            <Provider store={store}>
                <MemoryRouter>
                    <NavBtnsList />
                </MemoryRouter>
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});