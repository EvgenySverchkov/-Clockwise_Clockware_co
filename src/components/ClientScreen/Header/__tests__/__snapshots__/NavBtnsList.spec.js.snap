// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <NavBtnsList /> If user is logged 1`] = `
Array [
  <button
    aria-controls="navbarsExample03"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    data-target="#navbarsExample03"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>,
  <div
    className="navbar-collapse collapse"
    id="navbarsExample03"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/client"
          onClick={[Function]}
        >
          To main
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/client/userProfile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
    </ul>
  </div>,
]
`;

exports[`Test of <NavBtnsList /> If user is unlogged 1`] = `
Array [
  <button
    aria-controls="navbarsExample03"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    data-target="#navbarsExample03"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>,
  <div
    className="navbar-collapse collapse"
    id="navbarsExample03"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/client/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/client/registration"
          onClick={[Function]}
        >
          SignUp
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/client"
          onClick={[Function]}
        >
          To main
        </a>
      </li>
    </ul>
  </div>,
]
`;
